EzMap.Api/Controllers/PoiCollectionController.cs

```diff
@@ -16,13 +19,46 @@ public class PoiCollectionController : ControllerBase
     [HttpPost("")]
     public async Task<IActionResult> AddPoiCollection([FromBody] PoiCollectionCreateDto dto,
         [FromServices] IUnitOfWork uow,
-        [FromServices] IIdentityService identityService)
+        [FromServices] IIdentityService identityService, [FromServices] IElasticSearchService elasticSearchService)
     {
-        uow.PoiCollectionRepository.AddPoiCollection(dto.WithUserId(identityService.GetUserId()));
+        await using var transaction = await uow.BeginTransactionAsync();
+        try
+        {
+            var poiCollectionId =
+                uow.PoiCollectionRepository.AddPoiCollection(dto.WithUserId(identityService.GetUserId()));
+            var dbResult = await uow.SaveAsync();
+            if (dbResult <= 0)
+            {
+                await transaction.RollbackAsync();
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+
+            var poiCollectionCreateIndex =
+                new PoiCollectionCreateIndexingModel(poiCollectionId, dto.Name, dto.Description);
+            var esResult = await elasticSearchService.AddOrUpdate(poiCollectionCreateIndex);
+            if (!esResult)
+            {
+                await transaction.RollbackAsync();
+                return StatusCode(StatusCodes.Status500InternalServerError, new
+                {
+                    Message = "Failed to index the POI collection. The operation has been rolled back.",
+                    name = dto.Name
+                });
+            }
 
-        return await uow.SaveAsync() > 0
-            ? Ok("Your poi collection is created successfully!")
-            : new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            await transaction.CommitAsync();
+            return Ok(new
+            {
+                Message = "Your poi collection is created and indexed successfully!",
```

Comment:
Most 200 OK statuses don't need messages. The message shouldn't also leak implementation details (indexing), make it short, concise and minimal as possible. For error messages, try to also be concise, but add extra info on what's actually happening.

Reply:
It's also really inconsistent, this Ok returns a structured message: { Message, Name, Id }, while other Ok requests return only a plain string.